widgets/partials/public-monk-language-switcher.php

```diff
@@ -16,69 +16,70 @@
+		'fr_FR' => __( 'French', 'monk' ),
+		'pt_BR' => __( 'Portuguese', 'monk' ),
 	);
 }
 
 $languages_flag = array(
-	'pt-br' => 'http://image.flaticon.com/icons/svg/206/206597.svg',
-	'en'    => 'http://image.flaticon.com/icons/svg/206/206626.svg',
-	'es'    => 'http://image.flaticon.com/icons/svg/206/206724.svg',
-	'fr'    => 'http://image.flaticon.com/icons/svg/206/206657.svg',
+	'en_EN' => 'http://image.flaticon.com/icons/svg/206/206626.svg',
+	'es_ES' => 'http://image.flaticon.com/icons/svg/206/206724.svg',
+	'fr_FR' => 'http://image.flaticon.com/icons/svg/206/206657.svg',
+	'pt_BR' => 'http://image.flaticon.com/icons/svg/206/206597.svg',
 );
-
-/**
- * Show the flag in monk language switcher on public view
- *
- * @param bool  $flag
- * @param array $flag_list
- */
-function insert_flags( $flag, $flag_list ) {
-	if ( $flag ) {
-		return 'data-style="background-image: url( ' . $flag_list . ' );"';
-	} else {
-		return 'data-style="background-image: none;';
-	}
-};
-
-/**
- * Show the name of language in monk language switcher on public view
- *
- * @param bool  $name
- */
-function insert_names( $name ) {
-	if ( ! $name ) {
-		return 'screen-reader-text';	
-	}
-};
 ?>
-<form name="form-language" id="monk-form-language" method="get" action="<?php home_url(); ?>">
-	<select id="monk-widget-language-selector" name="lang" value="<?php echo $select_value; ?>">
-		<?php if ( isset( $select_value ) ) : ?>
-			<option data-class="monk-widget-option <?php echo insert_names( $name ); ?>" <?php echo insert_flags( $flag, $languages_flag[$select_value] ); ?> value="<?php echo esc_attr_e( $select_value, 'monk' ); ?>"><?php echo $languages[$select_value]; ?></option>
-		<?php endif; ?>
-	<?php foreach ( $languages as $key => $value ) : ?>
-		<?php if ( strcmp( $key, $select_value ) != 0 ) : ?>
-			<option data-class="monk-widget-option <?php echo insert_names( $name ); ?>" <?php echo insert_flags( $flag, $languages_flag[$key] ); ?> value="<?php echo esc_attr_e( $key, 'monk' ); ?>"><?php echo $value; ?></option>
-		<?php endif; ?>
-	<?php endforeach; ?>
-	</select>
-</form>
+<section id="<?php echo $this->id; ?>" class="widget <?php echo $this->id_base; ?>">
+	<h2 class="widget-title"><?php esc_html_e( $title ); ?></h2>
+	<ul id="monk-selector">
+		<?php foreach ( $languages as $key => $value ) : ?>
+			<?php if ( strcmp( $key, $select_value ) == 0 ) : ?>
+				<li class="monk-active-lang">
+					<span class="monk-active-lang-name">
+						<?php if ( $flag ) : ?>
+							<img class="monk-selector-flag" src="<?php esc_attr_e( $languages_flag[$key] ) ?>" alt="">
+						<?php endif; ?>
+						<?php if ( $name ) : ?>
+							<span><?php esc_html_e( $value ); ?></span>
+						<?php else : ?>
+							<span class="screen-reader-text"><?php esc_html_e( $value ); ?></span>
+						<?php endif; ?>
+					</span>
+				</li>
+			<?php else : ?>
+				<li class="monk-lang">
+					<a class="monk-selector-link" href="?lang=<?php esc_attr_e( $key, 'monk' ); ?>">
```

Comment:
Use `add_query_arg` function to generate the URL.
https://developer.wordpress.org/reference/functions/add_query_arg/